smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/PaginationGenerator.java

```diff
@@ -123,7 +123,7 @@ static void generateServicePaginationInterfaces(
 
      private String destructurePath(String path) {
         String[] splitIndex = path.split("\\.");
-        return "['" + String.join("']['", splitIndex) + "']";
+        return "." + String.join("?.", splitIndex);
```

Comment:
```suggestion
        return "."  + path.replace(".", "?.");
```

Reply:
Findings from offline meeting: The bang operator is used as optional chaining can't be used on the left hand side.

TS Playground: https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8UDeAoKUCGB+AXFAQigL4DcKKokBCyaUARngM7ABOAlgHYDmx5AxgHtOLKIPoArPHERJSKcRIB02JfRoAifABkYACQ0kgA